common/thrift_router.h

```diff
@@ -481,6 +496,10 @@ class ThriftRouter {
       }
     }
 
+    bool matchAZ(const std::string& host_az, const std::string& specific_az) {
+       return (specific_az == "" || host_az.compare(specific_az) == 0);
```

Comment:
Is there a reason we have the `specific_az == ""` check? The next check should take care of it, right?

Reply:
the `specific_az == ""` check is so we ignore checking AZ when it's not specified, i.e. always returning `true` here. 

Maybe the function can be renamed to make it clearer, how about `matchAZIfNotEmpty`

Reply:
compare returns a non zero value if specific_az is "". Hence we need the first condition. Basically, if specific_az is set to "", we do not have to do string compare.